Add LS release note (#2022)
ballerina-platform/ballerina-release: release-notes/swan-lake-release-notes/swan-lake-2201.1.1-release-note.md

To view bug fixes, see the [GitHub milestone for Swan Lake 2201.1.1]().

> ### Developer Tools Updates

Reviewer: ```suggestion
### Developer tools updates
```

> #### New Features

Reviewer: ```suggestion
#### New features
```

##### Language Server

> - Implemented document sync event subscriber for extensions

Reviewer 2: ```suggestion
- Implemented a publisher subscriber model for internal components and extensions to be notified about various events like project updates
```

> - Added code action support for records; to generate record types and to extract anonymous records into record types

Reviewer 2: ```suggestion
- Added new code action to extract anonymous records into records and generate undefined record types
```

- Introduced new code actions to generate getters and setters for class level variables

> - Added a new code action to make annotation declaration with 'source' attach point(s) constant

Reviewer: ```suggestion
- Added a new code action to make annotation declarations with the 'source' attach point(s) constant
```

#### Improvements

##### Language Server

- Improved hover feature support to display default values for defaultable parameters when hovering over a function
- Improved the `Create function` code action to add the isolated keyword to a function
- Improved the `Create function` code action to handle named arguments

> - Improved `Change return type` code action to add optional error for check expression

Reviewer: ```suggestion
- Improved the `Change return type` code action to add optional error for check expression
```

- Extended the `Type Cast` code action to support several unsupported binary operations

> - Improved completion item sorting in wait actions and already completed statements

Reviewer: ```suggestion
- Improved the completion item sorting in wait actions and already-completed statements
```

> - Updated the resource function snippet to add default values for resource method and path

Reviewer: ```suggestion
- Updated the resource function snippet to add default values for the resource method and path
```

> - Refactored the CommonUtil class by introducing several new utilities for the methods that are not used only by a single entity

Reviewer: ```suggestion
- Refactored the `CommonUtil` class by introducing several new utilities for the methods that are not used by only a single entity
```

- Modified the snippet labels for record type definitions
- Improved renaming support to avoid renaming on a compilation error

> - Improved strand annotation completion support in variable declarations

Reviewer: ```suggestion
- Improved the strand annotation completion support in variable declarations
```
